### PR TITLE
update go, actions, README and refactor tests

### PR DESCRIPTION
**Pull request by lufia**
I've updated them.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,43 +5,11 @@ on:
...
-        parallel: true
-        job-number: ${{ strategy.job-index }}
-  finish:
-    needs: [lint, test]
-    runs-on: ubuntu-latest
-    steps:
-    - uses: shogo82148/actions-goveralls@v1
-      with:
-        parallel-finished: true
+    use mackerelio/workflows/.github/workflows/go-test.yml@main
```
Comment (yseto) [inline .github/workflows/ci.yml]:
```suggestion
    - uses: mackerelio/workflows/.github/workflows/go-test.yml@main
```

File: README.md [hunk 2]
```diff
@@ -4,12 +4,12 @@ checkers
  [![Build Status](https://github.com/mackerelio/checkers/workflows/Build/badge.svg?branch=master)][actions]
 [![Coverage Status](https://coveralls.io/repos/mackerelio/checkers/badge.svg?branch=master)][coveralls]
 [![Apache License Version 2.0](http://img.shields.io/badge/license-APACHE2-blue.svg)][license]
```
Comment (itchyny) [inline README.md]: This badge link can use https.

**Review by yseto — APPROVED** [review]
seems good.

File: README.md (hunk 2 above)
Comment (lufia) [reply README.md]: I did! [f9f921d](https://github.com/mackerelio/checkers/pull/16/commits/f9f921dde8485b7a2cf34732754ebfdc39091409)

**Pull request merged**